### PR TITLE
Web tidy sweep

### LINKED ISSUE #310: Add ability to tag tasks as `unanswerable`
It's very likely we'll create a task that can't be answered because it's not sensible or complete garbage.

We should add a button and API piping to ensure a user can mark any task (not individual messages but the task itself) as *not answerable*.  This should ultimately  prevent the task being served to other users (perhaps after some threshold of users mark it as unanswerable) and provide negative feedback to whomever created the messages in the task.

### LINKED ISSUE #369: Going to the webpage should redirect to `/dashboard` for signin users
Right now if a user goes to the main website, it goes to the signed out page which doesn't have any obvious call to actions for contributing.  They can click the "dashboard" link in the user dropdown but it's not totally obvious.

Fixes might be:
- Have an obvious call to action for signed in users
- Redirect to `/dashboard` for signed in users

### LINKED ISSUE #371: Submitting tags on the website doesn't work due to the API route not existing
When you go to a task and try to flag a message with one of the checkbox entries, nothing happens.  Inspecting the network logs it's calling an API path (`/api/v1/text_labels`) that doesn't exist.

To fix this it requires:
- [ ] Making a new web api route under `website/src/pages/api/set_label.tsx`
- [ ] Update the component `website/src/components/FlaggableElement.tsx` to call this web API route
- [ ] Implement the web API route to forward the tag content to the backend server using the API key

### LINKED ISSUE #587: flaggable_labels is undefined rank_user_replies
This is a fall out of the recent fix on https://github.com/LAION-AI/Open-Assistant/issues/493

### PR DESCRIPTION
**Pull request by fozziethebeat**
Closes #369
Closes #310
Closes #371
Closes #587

Another sweep of tidying fixes:
- Cleanup more eslint warnings
- Update the `FlaggableElement` to call `off()` when reporting is done
- Update the `MessageView` to be forwardRef which is required by `PopoverAnchor`
- Update the reject task handler to use both `prisma` and `Prisma`
- Ensure that `FlaggableElement` has valid labels in every call site.

### CODE REVIEW

File: website/src/components/Messages/MessageTable.tsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { Stack, StackDivider } from "@chakra-ui/react";
 import { MessageTableEntry } from "src/components/Messages/MessageTableEntry";
 
-export function MessageTable({ messages }) {
+export function MessageTable({ messages, valid_labels }) {
+  console.log(messages);
```
Comment (AbdBarho) [inline website/src/components/Messages/MessageTable.tsx:5]: oh oh

**Review by AbdBarho — COMMENTED** [review]
This PR will have a lot of conflicts with #575 

should we merge that one first or this one?

**Comment by fozziethebeat** [comment]
> 

I say merge your PR first then I'll handle the conflicts here.  They might not actually be that bad.

**Review by k-nearest-neighbor — APPROVED** [review]
Runs & LGTM

File: website/src/components/Messages/MessageTable.tsx (hunk 1 above)
Comment (fozziethebeat) [reply website/src/components/Messages/MessageTable.tsx]: Removed!

**Comment by fozziethebeat** [comment]
I'll merge this now so that any new changes don't have to deal with potential conflicts.

**Pull request merged**